File: src/main/java/org/cerion/stocklist/web/Tools.java
package org.cerion.stocklist.web;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.net.MalformedURLException;
import java.net.URL;
import java.net.URLConnection;

public class Tools {

    public static String getURL(String theUrl)
    {
        String sResult = null;

        try
        {
            URL gotoUrl = new URL(theUrl);

            try (InputStreamReader isr = new InputStreamReader(gotoUrl.openStream());
                 BufferedReader in = new BufferedReader(isr))
            {
                StringBuffer sb = new StringBuffer();
                String inputLine;

                //grab the contents at the URL
                while ((inputLine = in.readLine()) != null)
                    sb.append(inputLine+"\r\n");

                sResult = sb.toString();
            }
            catch (IOException e)
            {
                sResult = "";
            }
        }
        catch (MalformedURLException mue) {
            mue.printStackTrace();
        }

        return sResult;
    }


    public static Response getURL(String url, String cookie)
    {
        Response result = new Response();

        try
        {
            URL gotoUrl = new URL(url);
            URLConnection conn = gotoUrl.openConnection();

            if (cookie != null)
                conn.setRequestProperty("Cookie", cookie);

            conn.connect();
            BufferedReader in = new BufferedReader(new InputStreamReader(conn.getInputStream()));

            StringBuffer sb = new StringBuffer();
            //DataInputStream dis = new DataInputStream(conn.getInputStream());

            result.headers = conn.getHeaderFields();

            String data;
            while ((data = in.readLine()) != null) {
                // System.out.println(urlData);
                if (sb.length() > 0)
                    sb.append("\r\n" + data);
                else
                    sb.append(data);
            }

            result.result = sb.toString();
        }
        catch (MalformedURLException mue)
        {
            mue.printStackTrace();
        }
        catch (IOException e)
        {
            result.result = "";
        }

        return result;
    }
}

File: src/main/kotlin/web/Tools.kt
package org.cerion.stocks.core.web

import java.io.BufferedReader
import java.io.IOException
import java.io.InputStreamReader
import java.net.HttpURLConnection
import java.net.MalformedURLException
import java.net.URL

object Tools {

    fun getUrl(urlString: String, headers: Map<String, String>?): Response {
        val res = Response()

        try {
            //Create connection
            val url = URL(urlString)
            val connection = url.openConnection() as HttpURLConnection
            connection.requestMethod = "GET"
            connection.useCaches = false
            connection.doInput = true

            headers?.forEach { connection.setRequestProperty(it.key, it.value) }
            //connection.setRequestProperty("Content-type", "application/json")
            //connection.setRequestProperty("Authorization", "Token ${Tiingo.API_TOKEN}")

            //Get Response
            res.code = connection.responseCode
            res.headers = connection.headerFields
            val stream = connection.inputStream
            val rd = BufferedReader(InputStreamReader(stream))
            var line: String?
            val response = StringBuffer()
            while (true) {
                line = rd.readLine()
                if (line == null)
                    break

                response.append(line)
                response.append('\r')
            }
            rd.close()
            res.result = response.toString()

        } catch (e: Exception) {
            //println("Response code = $code")
            //e.printStackTrace()
        } finally {
            //connection?.disconnect()
        }

        return res
    }

    fun getURL(theUrl: String): String? {
        var sResult: String? = null

        try {
            val gotoUrl = URL(theUrl)

            try {
                InputStreamReader(gotoUrl.openStream()).use { isr ->
                    BufferedReader(isr).use { input ->
                        val sb = StringBuffer()

                        //grab the contents at the URL
                        for (line in input.lines())
                            sb.append(line + "\r\n")

                        sResult = sb.toString()
                    }
                }
            } catch (e: IOException) {
                sResult = ""
            }

        } catch (mue: MalformedURLException) {
            mue.printStackTrace()
        }

        return sResult
    }


    fun getURL(url: String, cookie: String?): Response {
        val result = Response()

        try {
            val gotoUrl = URL(url)
            val conn = gotoUrl.openConnection()

            if (cookie != null)
                conn.setRequestProperty("Cookie", cookie)

            conn.connect()
            val reader = BufferedReader(InputStreamReader(conn.getInputStream()))

            val sb = StringBuffer()
            //DataInputStream dis = new DataInputStream(conn.getInputStream());

            result.headers = conn.headerFields

            while (true) {
                val line = reader.readLine()
                if (line != null)
                    sb.append(line + "\r\n")
                else
                    break
            }

            result.result = sb.toString()
        } catch (mue: MalformedURLException) {
            mue.printStackTrace()
        } catch (e: IOException) {
            result.result = ""
        }

        return result
    }
}
